src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -556,9 +557,7 @@ private static void ThrowConcurrentExceptionOnMissingDoc(string id, string expec
 
         private static void ThrowInvalidCollectionNameChange(string id, CollectionName oldCollectionName, CollectionName collectionName)
         {
-            throw new InvalidOperationException(
-                $"Changing '{id}' from '{oldCollectionName.Name}' to '{collectionName.Name}' via update is not supported.{Environment.NewLine}" +
-                $"Delete it and recreate the document {id}.");
+            throw new CollectionMismatchException(id, oldCollectionName.Name, collectionName.Name);
```

Comment:
switch to static factory 'ThrowFor' and put in inside the ex class e.g. like in IndexDoesNotExistException